src/server/utilities/constructCspHeader/index.test.js

```diff
@@ -0,0 +1,276 @@
+      ],
+      false,
+      true,
+      true,
+    );
+  });
+
+  it('should be able to generate the test connect src when in the uk on cannonical', async () => {
+    testConnect(
+      [
+        'https://logws1363.ati-host.net',
+        "'self'",
+        'https://cookie-oven.api.bbc.co.uk',
+        'https://cookie-oven.test.api.bbc.co.uk',
+      ],
+      false,
+      true,
+      false,
+    );
+  });
+
+  it('should be able to generate the live connect src when not in the uk on cannonical', async () => {
+    testConnect(
+      [
+        'https://a1.api.bbc.co.uk/hit.xiti',
+        "'self'",
+        'https://cookie-oven.api.bbc.com',
+      ],
+      false,
+      false,
+      true,
+    );
+  });
+
+  it('should be able to generate the test connect src when not in the uk on cannonical', async () => {
+    testConnect(
+      [
+        'https://logws1363.ati-host.net',
+        "'self'",
+        'https://cookie-oven.api.bbc.com',
+        'https://cookie-oven.test.api.bbc.com',
+      ],
+      false,
+      false,
+      false,
+    );
+  });
+
+  it('should be able to generate the test connect src when not in the uk on amp', async () => {
+    testConnect(['https://logws1363.ati-host.net'], true, false, false);
+  });
+
+  it('should be able to generate the live connect src when not in the uk on amp', async () => {
+    testConnect(['https://a1.api.bbc.co.uk/hit.xiti'], true, false, true);
+  });
+
+  it('should be able to generate the test connect src when in the uk on amp', async () => {
+    testConnect(['https://logws1363.ati-host.net'], true, true, false);
+  });
+
+  it('should be able to generate the live connect src when in the uk on amp', async () => {
+    testConnect(['https://a1.api.bbc.co.uk/hit.xiti'], true, true, true);
+  });
+});
+
+describe('CSP Header Middleware', () => {
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('should be able to inject the csp header for live in the uk on cannon', async () => {
+    testMiddleware(
+      'https://bbc.co.uk',
+      '/igbo',
+      "default-src 'self'; font-src https://gel.files.bbci.co.uk https://ws-downloads.files.bbci.co.uk; style-src 'unsafe-inline'; img-src https://ichef.bbci.co.uk https://ping.chartbeat.net https://a1.api.bbc.co.uk/hit.xiti https://news.files.bbci.co.uk https://*.akstat.io https://r.bbci.co.uk data: 'self'; script-src https://news.files.bbci.co.uk https://*.chartbeat.com https://*.go-mpulse.net https://mybbc-analytics.files.bbci.co.uk https://emp.bbci.co.uk https://static.bbci.co.uk 'self' 'unsafe-inline'; connect-src https://*.akstat.io https://*.akamaihd.net https://c.go-mpulse.net https://a1.api.bbc.co.uk/hit.xiti 'self' https://cookie-oven.api.bbc.co.uk; frame-src 'self' https://emp.bbc.com https://emp.bbc.co.uk https://chartbeat.com https://*.chartbeat.com",
+    );
+  });
+
+  it('should be able to inject the csp header for live in the uk on amp', async () => {
+    testMiddleware(
```

Comment:
to make this a little more readable it might be a good idea to pass in an object rather than multiple args

Reply:
Yep on it